src/core/qgsvectordataprovider.h

```diff
@@ -533,6 +543,13 @@ class CORE_EXPORT QgsVectorDataProvider : public QgsDataProvider, public QgsFeat
      */
     void raiseError( const QString &msg ) const;
 
+    /**
+     * Notification coming from provider
+     *
+     * \since QGIS 3.0
+     */
+    void notify( QString msg ) const;
```

Comment:
Shouldn't `msg` be const?

Reply:
Here it's "pass by value". But I'm not sure what is the best for signals since it's more a return value than anything else, the signal interface is misleading IMO.

Apparently Qt does some magic such that a "const &" is actually transformed into a "pass by value" https://stackoverflow.com/questions/8455887/stack-object-qt-signal-and-parameter-as-reference

Reply:
It should be const ref, even in signals.

The Qt magic related to old style connections - but we don't use them anymore.

Reply:
Fair enough. Done.